### DISCUSSION TITLE
Mania scroll speed does not match the numbers in stable

### DISCUSSION THREAD

**Discussion by Morilli** [ppy/osu#13896] [Q&A] [answered]
**Describe the bug:** 
I noticed that the displayed scroll speed number (1 - 40) does not match the scroll speed of stable. I have no idea what the "correct" math is but I don't think I'm able to find out easily.

Example:
Scroll speed of 13720ms (1, the lowest you can go) in lazer results in ~10000 milliseconds time between seeing the note for the first time and being able to hit it, whereas a scroll speed of 1 in stable gives ~11500 milliseconds. None of these numbers make any sense to me given that it says 13720ms.

Another example for the opposite case, aka 340ms (40):
Lazer time between seeing the note and hitting it: about 20 frames on my 60fps recording, aka ~333ms.
Stable time: 17 frames on a 60fps recording, so around 283ms.

I was timing from the first frame any part of the note was visible to the frame in which autoplay hits the note, in the standard skin.

**osu!lazer version:** 
2021.707.0

**Comment by peppy** [comment]
Moving to discussion, as this was done mostly intentionally (ie. it matches closely but I believe is done in a more sane way here, vs stable). We'll probably want to migrate people's values across to give a starting point, but this requires further discussion.

**Comment by smoogipoo** [comment]
Lazer's 13720ms should absolutely be 13720ms. Are you sure you're not just seeing an SV-adjusted value?

Can you provide a video comparison?

In theory, all that's required is to figure out the time for stable at speeds 1 and 40 to replace the existing values. But I don't know how you've tested this, because we've calculated the current values numerically in https://github.com/ppy/osu/issues/9868.

**Reply by Morilli** [reply]
The map I'm testing with is [Queenland's Clarity (Another World)](https://osu.ppy.sh/beatmapsets/485725#mania/1035769), which does not have any SVs at all.

Here's a video of the map in stable with maximum scroll speed: https://youtu.be/NHYW3eqNvtk
And here in lazer: https://youtu.be/ANfYhCOhFOM

I don't have videos for scroll speed of 1 right now, but the difference should be immediately visible just from note density.

As I said I just timed the amount of frames between note visible -> note gets hit in a 60fps recording; I haven't done any math.

**Comment by smoogipoo** [comment]
Hmm... So I tested this locally and confirmed one issue. In the original calculations (https://github.com/ppy/osu/issues/9868#issuecomment-675327188) I assumed a scrolling distance of 480, but this isn't actually correct. It should be 402.

Which means that a more accurate time range is:
```
MIN_TIME_RANGE = 290
MAX_TIME_RANGE = 11485
```
Can you test with these values locally and see if the difference in speed is noticeable to you still across several maps?

Here's a comparison:
| stable | lazer |
| --- | --- |
| ![Screenshot_2021-07-16_17-11-54](https://user-images.githubusercontent.com/1329837/125923075-1c44924d-4892-456b-9cde-ff5d050210d3.png) | ![Screenshot_2021-07-16_17-12-44](https://user-images.githubusercontent.com/1329837/125923089-f01bd49a-06a3-4e11-8a56-ff2e0d81b34a.png) |
| ![Screenshot_2021-07-16_17-14-36](https://user-images.githubusercontent.com/1329837/125923109-7d5e19fe-1b36-477f-a5c0-ecca97fae254.png) | ![Screenshot_2021-07-16_17-19-17](https://user-images.githubusercontent.com/1329837/125923120-0a6bc23a-ae19-409b-9bcd-aeb893c92206.png) |

I have no idea where the rest of the differences are coming from...

**Comment by smoogipoo** [comment, marked as answer]
Have resolved in https://github.com/ppy/osu/pull/13901
